iterableapi/src/main/java/com/iterable/iterableapi/IterableActionReceiver.java

```
@@ -1,62 +1,58 @@
 package com.iterable.iterableapi;
 
-import android.app.Activity;
 import android.app.NotificationManager;
+import android.content.BroadcastReceiver;
 import android.content.Context;
 import android.content.Intent;
 import android.net.Uri;
 import android.os.Bundle;
 
-/**
- * Created by David Truong dt@iterable.com.
- */
-
-public class IterableLaunchActivity extends Activity {
+public class IterableActionReceiver extends BroadcastReceiver {
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-
-        Intent intent = getIntent();
+    public void onReceive(Context context, Intent intent) {
         Bundle extras = intent.getExtras();
 
         //TOOD: might not need the extra requestCode since the messageId is always in the original payload.
         int requestCode = extras.getInt(IterableConstants.REQUEST_CODE, 0);
         System.out.print("requestCode: "+requestCode);
         NotificationManager mNotificationManager = (NotificationManager)
-                this.getBaseContext().getSystemService(Context.NOTIFICATION_SERVICE);
+                context.getSystemService(Context.NOTIFICATION_SERVICE);
         mNotificationManager.cancel(requestCode);
 
-        Class mainClass = (Class) extras.get(IterableConstants.MAIN_CLASS);
+        Class mainClass = null;
+        try {
+            mainClass = Class.forName(extras.getString(IterableConstants.MAIN_CLASS, ""));
+        } catch (ClassNotFoundException e) {
+            IterableLogger.e("IterableActionReceiver", "Invalid main class", e);
```

If mainClass can't be loaded should probably throw a runtime exception after log.